server/models/board.js

```diff
@@ -0,0 +1,55 @@
+import mongoose from 'mongoose';
+
+const { Schema } = mongoose;
+
+const BoardSchema = new Schema({
+  name: {
+    type: String,
+    trim: true,
+    required: true,
+  },
+  createdBy: {
+    type: Schema.Types.ObjectId,
+    ref: 'User',
+    required: true,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+    required: true,
+  },
+  members: [
+    {
+      member: {
+        type: Schema.Types.ObjectId,
+        ref: 'User',
+      },
+      role: {
+        type: String,
+        enum: ['SUPERADMIN', 'ADMIN', 'USER', 'INVITED'],
+        default: 'USER',
+        required: true,
+      },
+    },
+  ],
+  lifeCycles: [
+    {
+      name: {
```

Comment:
create another schema as lifeCycles and use all fields over there, just add here references

Reply:
I didn't take the reference path for _lifecycles_ because one board will usually have 3-5 lifecycles (7 at the most for sane use cases). We can even put an upper limit (10?) on the number of lifecycles in a board.
Secondly, whenever we'll need board details we'll always have to know its lifecycles and having lifecycles embedded here would be easier to query. 
Let me know your thoughts.